modules/dxp/apps/search-experiences/search-experiences-web/src/test/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommandTest.java

```diff
@@ -0,0 +1,397 @@
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 2, sitesJSONArray.length());
+	}
+
+	@Test
+	public void testGetSitesJSONObjectWithDifferentCompanyGroupId()
+		throws Exception {
+
+		_setUpGroups(false, 3, new ArrayList<>());
+
+		JSONObject sitesJSONObject =
+			_getSitesMVCResourceCommand.getSitesJSONObject(_resourceRequest);
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 0, sitesJSONArray.length());
+	}
+
+	@Test
+	public void testGetSitesJSONObjectWithSameCompanyGroupId()
+		throws Exception {
+
+		_setUpPages(_resourceRequest, 10, 1);
+
+		_setUpGroups(true, 3, new ArrayList<>());
+
+		JSONObject sitesJSONObject =
+			_getSitesMVCResourceCommand.getSitesJSONObject(_resourceRequest);
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 3, sitesJSONArray.length());
+	}
+
+	@Test
+	public void testServeResource() throws Exception {
+		_setUpResourceRequest();
+		_setUpResourceResponse();
+
+		_setUpParamUtil(Constants.CMD, "getSitesJSONObject");
+
+		_getSitesMVCResourceCommand.serveResource(
+			_resourceRequest, _resourceResponse);
+
+		Mockito.verify(
+			_resourceResponse, Mockito.times(1)
+		).isCommitted();
+	}
+
+	@Test
+	public void testSitesJSONObjectPaginationConditions() throws Exception {
+		_setUpPages(_resourceRequest, 5, 3);
+		_setUpGroups(true, 8, new ArrayList<>());
+
+		JSONObject sitesJSONObject =
+			_getSitesMVCResourceCommand.getSitesJSONObject(_resourceRequest);
+
+		Assert.assertEquals(2, sitesJSONObject.getInt("lastPage"));
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 3, sitesJSONArray.length());
+	}
+
+	@Test
+	public void testSitesJSONObjectPaginationOnePage() throws Exception {
+		_setUpPages(_resourceRequest, 10, 1);
+		_setUpGroups(true, 8, new ArrayList<>());
+
+		JSONObject sitesJSONObject =
+			_getSitesMVCResourceCommand.getSitesJSONObject(_resourceRequest);
+
+		Assert.assertEquals(1, sitesJSONObject.getInt("lastPage"));
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 8, sitesJSONArray.length());
+	}
+
+	@Test
+	public void testSitesJSONObjectPaginationTwoPages() throws Exception {
+		_setUpPages(_resourceRequest, 5, 2);
+		_setUpGroups(true, 10, new ArrayList<>());
+
+		JSONObject sitesJSONObject =
+			_getSitesMVCResourceCommand.getSitesJSONObject(_resourceRequest);
+
+		Assert.assertEquals(2, sitesJSONObject.getInt("lastPage"));
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 5, sitesJSONArray.length());
+	}
+
+	private Group _createGroup(
+			String descriptiveName, boolean fromCompanyGroupId,
```

Comment:
`fromCompanyGroupId` all through this test and in the parallel test needs to be renamed to something like `isActive`.

Reply:
as a note the SF complies about `isActive` the pattern is just `active`